apps/src/music/views/ChordPanel.tsx

```diff
@@ -131,21 +102,18 @@ const ChordPanel: React.FunctionComponent<ChordPanelProps> = ({
         onPressKey={onPressKey}
         isDisabled={isDisabled}
       />
-      <div className={moduleStyles.controlsRow}>
-        <FontAwesome
-          icon={'volume-up'}
-          onClick={onPreview}
-          className={classNames(
-            moduleStyles.previewButton,
-            isPlayingPreview && moduleStyles.previewButtonDisabled
-          )}
-        />
-        <FontAwesome
-          icon={'ban'}
-          onClick={() => setSelectedNotes([])}
-          className={moduleStyles.previewButton}
-        />
-      </div>
+      <PreviewControls
+        enabled={selectedNotes.length > 0}
+        playPreview={() =>
```

Comment:
I guess it's rather general comment than a comment to a specific PR. (Looks like there're more then 1 place where this can be applied to)

I see that we use a functional component with hooks here. I'm curious if there is any reason why we don't use `useCallback` here for optimisation?
I remember we had this type of issue in older lab in cdo that I had to fix. It was using React.Component API so the fix in that case was a little bit different, but if we were using functional components with hooks at that place, `useCallback` would simply prevent appear of that issue.

A small article about `useCallback` in case anyone will be interested: [](https://www.knowledgehut.com/blog/web-development/all-about-react-usecallback)

Reply:
Good question! Honestly, I think we're just not really in the habit of using `useCallback` but we should be doing so more. I'll update these and it's something we can definitely plan to do more of going forward.